### PR TITLE
Make slot prediction logic public

### PR DESCRIPTION
**Pull request by 4meta5**
as discussed in this conversation https://github.com/PureStake/moonbeam/pull/744#discussion_r699682255

### CODE REVIEW

File: pallets/author-slot-filter/src/lib.rs [hunk 1]
```diff
@@ -61,46 +61,44 @@ pub mod pallet {
 		type PotentialAuthors: Get<Vec<Self::AccountId>>;
 	}
 
+	// then we call this from runtime APIs for author slot prediction
```
Comment (4meta5) [inline pallets/author-slot-filter/src/lib.rs:64]: I'll replace this with proper doc comment after the idea is OKed

**Review by JoshOrndorff — APPROVED** [review]
Looks good!

File: pallets/author-slot-filter/src/lib.rs [hunk 2]
```diff
@@ -32,8 +32,7 @@ pub use pallet::*;
 #[pallet]
 pub mod pallet {
 
-	use frame_support::pallet_prelude::*;
-	use frame_support::traits::Randomness;
+	use frame_support::{pallet_prelude::*, traits::Randomness};
```
Comment (4meta5) [inline pallets/author-slot-filter/src/lib.rs]: all the import changes were cargo fmt, I'll revert them before merging

File: pallets/author-slot-filter/src/lib.rs [hunk 3]
```diff
@@ -61,50 +60,52 @@ pub mod pallet {
...
-				// TODO we could short-circuit this check by returning early when the claimed
-				// author is selected. For now I'll leave it like this because:
-				// 1. it is easier to understand what our core filtering logic is
-				// 2. we currently show the entire filtered set in the debug event
-				eligible.push(active.remove(index % active.len()));
-			}
+			let active: Vec<T::AccountId> = T::PotentialAuthors::get();
+
+			// Compute pseudo-random subset of potential set
+			let (eligible, ineligible) = compute_pseudo_random_subset::<T>(active, slot);
```
Comment (4meta5) [inline pallets/author-slot-filter/src/lib.rs:104]:
```suggestion
			// Compute pseudo-random subset of potential set
			let (eligible, ineligible) = compute_pseudo_random_subset::<T>(T::PotentialAuthors::get(), slot);
```

Comment (4meta5) [reply pallets/author-slot-filter/src/lib.rs:104]: This is more of a code style question @JoshOrndorff , which do you prefer

Comment (4meta5) [reply pallets/author-slot-filter/src/lib.rs:104]: This makes me realize we can put the `get` inside the function. I'd prefer that actually. One sec, will push.

**Comment by JoshOrndorff** [comment]
TODO I need to cherry-pick this back into the `nimbus` branch. Will be easy.

**Pull request merged**